generator/cmd/root.go

```diff
@@ -67,8 +45,38 @@ func init() {
 	rootCmd.Flags().StringP(inputArg, "i", "", "path to the OpenAPI spec")
 	rootCmd.Flags().StringP(outputArg, "o", "", "destination to output the file to")
 	rootCmd.Flags().StringP(templateArg, "t", "", "path to the template file if defaults need overriding")
+	rootCmd.Flags().StringP(GenerateType, "g", "", "what to generate")
 
 	_ = configUtils.BindFlagToEnv(viperSession, app, "GENERATE_CODE_INPUT", rootCmd.Flags().Lookup(inputArg))
 	_ = configUtils.BindFlagToEnv(viperSession, app, "GENERATE_CODE_OUTPUT", rootCmd.Flags().Lookup(outputArg))
 	_ = configUtils.BindFlagToEnv(viperSession, app, "GENERATE_CODE_TEMPLATE", rootCmd.Flags().Lookup(templateArg))
+	_ = configUtils.BindFlagToEnv(viperSession, app, "GENERATE_CODE_GENERATE_TYPE", rootCmd.Flags().Lookup(GenerateType))
+}
+
+func RunCLI(ctx context.Context) (err error) {
+	logger := logrimp.NewStdOutLogr()
+
+	err = configUtils.LoadFromViper(viperSession, app, &extensionConfig, codegen.DefaultExtensionsConfig())
+	if err != nil {
+		logger.Error(err, "Failed to initialise CLI with error")
+		return
+	}
+
+	d, err := codegen.GenerateDataStruct(extensionConfig)
+	if err != nil {
+		return
+	}
+
+	generateParams, ok := codegen.ValidGenerationTargets[extensionConfig.GenerateType]
+	if ok {
+		err = generateParams(d)
+	} else {
+		err = fmt.Errorf("%w. '%s' is not a valid generation type", commonerrors.ErrNotFound, extensionConfig.GenerateType)
```

Comment:
we added utilities in golang-utils to ease the wrapping of errors `commonerrors.Newf(commonerrors.ErrNotFound, "'%s' is not a valid generation type", extensionConfig.GenerateType)` see https://github.com/ARM-software/golang-utils/blob/951d4c8417b84387444eb1b0e8f4bee3f4f20d78/utils/commonerrors/errors.go#L338